tests/test_inner.py

```diff
@@ -588,10 +606,9 @@ def test_workflow_plugin_error(fail_at):
     # earlier plugin failures.
     assert watch_post.was_called() == (fail_at not in ('pre', 'prepub', 'post'))
 
-    # But all exit plugins should run, even if one of them also raises
-    # an exception.
-    assert watch_exit.was_called()
-
+    # Because exit_raises is marked as is_allowed_to_fail=False, it prevents
+    # all other exit plugins to continue.
```

Comment:
I don't think this is quite what we need. We want all the exit plugins to run regardless of failure, but if one fails that is "not allowed to fail", the build as a whole should fail - after all the exit plugins have run.